device/mellanox/x86_64-mlnx_msn2700-r0/plugins/psuutil.py

```diff
@@ -8,22 +8,49 @@
 #
 #############################################################################
 
-import os.path
-
 try:
+    import os.path
+    import syslog
+    import subprocess
     from sonic_psu.psu_base import PsuBase
 except ImportError as e:
     raise ImportError (str(e) + "- required module not found")
 
+def log_err(msg):
+    syslog.openlog("psuutil")
+    syslog.syslog(syslog.LOG_ERR, msg)
+    syslog.closelog()
+
+
 class PsuUtil(PsuBase):
     """Platform-specific PSUutil class"""
 
+    MAX_PSU_FAN = 1
+    GET_HWSKU_CMD = "sonic-cfggen -d -v DEVICE_METADATA.localhost.hwsku"
+    # for spectrum1 switches with plugable PSUs, the output voltage file is psuX_volt
+    # for spectrum2 switches the output voltage file is psuX_volt_out2
+    sku_spectrum1_with_plugable_psu = ['ACS-MSN2410', 'ACS-MSN2700', 'Mellanox-SN2700', 'Mellanox-SN2700-D48C8', 'LS-SN2700', 'ACS-MSN2740']
```

Comment:
how about 3700, 3700C, 3800? maybe define a list for SKU with unplugable_psu?

Reply:
we don't need that.
1. we have SKUs with unpluggable PSU (2010, 2100) but we don't care about them becuase voltage, current and power aren't support by them.
2. There is a difference in voltage file name between SPC1 and SPC2 SKUs. To distinguish them a dictionary for Spectrum1 pluggable SKUs is defined.